src/main/java/com/uber/cadence/worker/WorkerOptions.java

```diff
@@ -17,112 +17,140 @@
 package com.uber.cadence.worker;
 
 import com.uber.cadence.converter.DataConverter;
+import com.uber.cadence.converter.JsonDataConverter;
 
 public final class WorkerOptions {
 
-    boolean disableWorkflowWorker;
-    boolean disableActivityWorker;
-    double workerActivitiesPerSecond;
-    String identity;
-    DataConverter dataConverter;
-    int maxConcurrentActivityExecutionSize;
-    int maxWorkflowThreads;
-
-    public int getMaxConcurrentActivityExecutionSize() {
-        return maxConcurrentActivityExecutionSize;
-    }
-
-    /**
-     * Maximum number of parallely executed activities.
-     */
-    public void setMaxConcurrentActivityExecutionSize(int maxConcurrentActivityExecutionSize) {
-        this.maxConcurrentActivityExecutionSize = maxConcurrentActivityExecutionSize;
-    }
-
-    public int getMaxWorkflowThreads() {
-        return maxWorkflowThreads;
-    }
-
-    /**
-     * Maximum size of a thread pool used to execute workflows.
-     */
-    public void setMaxWorkflowThreads(int maxWorkflowThreads) {
-        this.maxWorkflowThreads = maxWorkflowThreads;
-    }
-
-    public String getIdentity() {
-        return identity;
-    }
-
-    public void setIdentity(String identity) {
+    public final static class Builder {
+
+        boolean disableWorkflowWorker;
+        boolean disableActivityWorker;
+        double workerActivitiesPerSecond;
+        String identity;
+        DataConverter dataConverter = JsonDataConverter.getInstance();
+        int maxConcurrentActivityExecutionSize;
+        int maxWorkflowThreads;
+
+        /**
+         * When set to true doesn't poll on workflow task list even if there are registered workflows with a worker.
+         * For clarity prefer not registing workflow types with a {@link Worker} to setting this option.
+         * But it can be useful for disabling polling through configuration without a code change.
+         */
+        public Builder setDisableWorkflowWorker(boolean disableWorkflowWorker) {
```

Comment:
is there a reason to take a boolean parameter to this function ? setDisableWorkflowWorker() itself is self documenting